igcollect/ping.py

```diff
@@ -0,0 +1,122 @@
+#!/usr/bin/env python3
+
+"""igcollect - Latency collection
+Copyright (c) 2020 InnoGames GmbH
+"""
+
+# This script is used to collect ping data to evaluate availability and latency
+# of our network and make it easy to create a Graphite Dashboard
+#
+# Please note that the package fping has to be installed on your Linux server!
+
+
+import argparse
+import subprocess
+import time
+
+from multiprocessing import Pool
+from os import cpu_count
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description='Ping collector script for Graphite ')
+    parser.add_argument('hosts', nargs='+',
+                        help='the hosts to ping')
+    parser.add_argument('-c', '--count', type=int, default=20, 
+                        choices=range(1,60),
+                        help='how many times the script will try to ping '
+                             '(number has to be bigger '
+                             'than 1 and smaller than 60)')
+    parser.add_argument('-p', '--prefix', default='ping',
+                        help='the path to the value in Graphite')
+    parser.add_argument('-t', '--timeout', type=int, default=300,
+                        help='time in milliseconds '
+                             'till the timeout is retched')
+    return parser.parse_args()
+
+
+def main():
+    args = parse_args()
+
+    chunked_hosts = [args.hosts[i:i + 10]
+                     for i in range(0, len(args.hosts), 10)]
+
+    data = [(args.prefix, chunk, args.count, args.timeout)
+            for chunk in chunked_hosts]
+
+    with Pool(cpu_count() * 2) as p:
+        p.starmap(check_pings, data)
+
+
+def check_pings(prefix, hosts, count, timeout):
+    values = pings(hosts, count, timeout)
+
+    timestamp = str(int(time.time()))
+
+    for data in values:
+        send(prefix, data, timestamp)
+    
+
+def pings(hosts, count, timeout):
+    hosts_string = " ".join(hosts)
+    # -p calculates the time fping waits between single ping probes
+    # with 20 probes and 1800 milliseconds in between it takes roughly 1 minute
+    # if you use 30 probes the -p time should goe down, for 10 up
+    cmd = 'fping -B1 -q -C {} -p {} -t {} {}'
+    cmd = cmd.format(count, (20/count) * 1800, timeout, hosts_string)
+
+    # example.com : 105.94 - 104.78
+    # Showing times for each ping. A ping timeout will be shown as '-'.
+    output = subprocess.getoutput(cmd).split('\n')
+
+    values = []
+    for line in output:
+        data = {}
+        parts = line.split(':', 1)
+        data['dest'] = parts[0].strip(' ').replace('.', '_')
+        pings = parts[1].split()
+
+        data['max'] = max(pings)
+        data['min'] = min(pings)
+        
+        total = 0
```

Comment:
As you are awaiting floats, make it already clear here:
```suggestion
        total = 0.0
```